inference-engine/src/mkldnn_plugin/nodes/mkldnn_adaptive_pooling.cpp

```diff
@@ -0,0 +1,360 @@
+    }
+}
+
+void MKLDNNAdaptivePoolingNode::initSupportedPrimitiveDescriptors() {
+    if (!supportedPrimitiveDescriptors.empty())
+        return;
+
+    if (mayiuse(avx512_core) && (outputPrecision == Precision::BF16 || inputPrecision == Precision::BF16)) {
+      outputPrecision = inputPrecision = Precision::BF16;
+    } else {
+      outputPrecision = inputPrecision = Precision::FP32;
+    }
+
+    auto inputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(inputPrecision);
+    auto outputDataType = inputDataType;
+
+    InferenceEngine::LayerConfig config;
+    config.dynBatchSupport = false;
+    config.inConfs.resize(2);
+    config.outConfs.resize((algorithm == Algorithm::AdaptivePoolingAvg ? 1 : 2));
+
+    for (auto fmt : getAvailableFormatsForDims(getParentEdgeAt(0)->getDims())) {
+        config.inConfs[0].desc = MKLDNNMemoryDesc(getParentEdgeAt(0)->getDims(), inputDataType, fmt);
+        config.inConfs[1].desc = MKLDNNMemoryDesc(getParentEdgeAt(1)->getDims(), memory::data_type::s32, memory::format_tag::x);
+        config.outConfs[0].desc = MKLDNNMemoryDesc(getChildEdgeAt(0)->getDims(), outputDataType, fmt);
+        if (algorithm == Algorithm::AdaptivePoolingMax) {
+            memory::format_tag indexFmt;
+            switch (spatialDimsCount) {
+                case 1:
+                    indexFmt =  memory::format_tag::ncw;
+                    break;
+                case 2:
+                    indexFmt = memory::format_tag::nchw;
+                    break;
+                case 3:
+                    indexFmt = memory::format_tag::ncdhw;
+                    break;
+                default:
+                    IE_THROW() << "Incorrect spatial dims count(" << spatialDimsCount << ")";
+            }
+            config.outConfs[1].desc = MKLDNNMemoryDesc(getChildEdgeAt(1)->getDims(), memory::data_type::s32, indexFmt);
+        }
+        supportedPrimitiveDescriptors.push_back({config, impl_desc_type::unknown, fmt});
+    }
+}
+
+std::vector<memory::format_tag>
+MKLDNNAdaptivePoolingNode::getAvailableFormatsForDims(const MKLDNNDims &dims) const {
+    auto plainFmt = [&]() {
+        if (dims.ndims() == 3)
+            return memory::format_tag::ncw;
+        else if (dims.ndims() == 4)
+            return memory::format_tag::nchw;
+        else if (dims.ndims() == 5)
+            return memory::format_tag::ncdhw;
+        return memory::format_tag::any;
+    };
+    auto blockedFmt = [&]() {
+        if (dims.ndims() == 3)
+            return mayiuse(avx512_common) ? memory::format_tag::nCw16c : memory::format_tag::nCw8c;
+        else if (dims.ndims() == 4)
+            return mayiuse(avx512_common) ? memory::format_tag::nChw16c : memory::format_tag::nChw8c;
+        else if (dims.ndims() == 5)
+            return mayiuse(avx512_common) ? memory::format_tag::nCdhw16c : memory::format_tag::nCdhw8c;
+        return memory::format_tag::any;
+    };
+    auto tailFmt = [&]() {
```

Comment:
Looks like these formats are not supported here.

Reply:
?